docs/cccl_development/macro.rst

```diff
@@ -144,6 +165,31 @@ The following macros are required only if the target C++ version does not suppor
 | ``_CCCL_INLINE_VAR``        | Portable ``inline constexpr`` variable (before C++17)    |
 +-----------------------------+----------------------------------------------------------+
 
+**Concept-like Macros**:
+
++------------------------+--------------------------------------------------+
+| ``_CCCL_TEMPLATE(X)``  | ``template`` clause                              |
++------------------------+--------------------------------------------------+
+| ``_CCCL_REQUIRES(X)``  | ``requires`` clause                              |
++------------------------+--------------------------------------------------+
+| ``_CCCL_TRAIT(X)``     | Template parameter constrain                     |
++------------------------+--------------------------------------------------+
+| ``_CCCL_AND``          | Traits conjunction                               |
++------------------------+--------------------------------------------------+
+
+Usage example:
+
+.. code-block:: c++
+
+    _CCCL_TEMPLATE(typename T)
+    _CCCL_REQUIRES(_CCCL_TRAIT(is_integral, T) _CCCL_AND(sizeof(T) > 1))
+
+.. code-block:: c++
+
+    _CCCL_TEMPLATE(typename T)
+    _CCCL_REQUIRES(_CCCL_TRAIT(is_arithmetic, T) _CCCL_AND !_CCCL_TRAIT(is_integral, T))
```

Comment:
This is actually not valid because of the !
```suggestion
    _CCCL_REQUIRES(_CCCL_TRAIT(is_arithmetic, T) _CCCL_AND (!_CCCL_TRAIT(is_integral, T)))
```